jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/servlets/SPARQL_Update.java

```diff
@@ -92,13 +93,15 @@ protected void doOptions(HttpServletRequest request, HttpServletResponse respons
         response.setHeader(HttpNames.hAllow, "OPTIONS,POST") ;
         response.setHeader(HttpNames.hContentLengh, "0") ;
     }
-    
+
     protected void doOptions(HttpAction action) {
         doOptions(action.request, action.response) ;
     }
 
     @Override
     protected void perform(HttpAction action) {
+        Cache cache = SPARQL_Query_Cache.getCache();
```

Comment:
A more subtle approach would be to clear the cache only if the action goes through. If it fails and no chance is made to the data, there's no need to clear the cache.